nrpy/helpers/safewrite.py

```diff
@@ -0,0 +1,87 @@
+"""
+The Safewrite class helps users see and understand changes to generated output files.
+
+Author: Steven R. Brandt
+"""
+
+from typing import Any, Optional, Union
+import io
+import os
+from difflib import context_diff
+import sys
+from subprocess import Popen, PIPE
+from pathlib import Path
+from nrpy.helpers.colored import colored
+
+try:
```

Comment:
Not necessary. clang-format is already a requirement. No need to double-check; it'll be handled by `clang_format()` in helpers/generic.py (see below comment)